doc/specs/#674 - PowerShell cmdlets.md

```diff
@@ -0,0 +1,434 @@
+
+```PowerShell
+Add-WinGetSource
+```
+
+```PowerShell
+Remove-WinGetSource
+```
+
+```PowerShell
+Get-WinGetSource
+```
+
+```PowerShell
+Reset-WinGetSource
+```
+
+**Example**
+> Note: `winget source export` was added as a helper for adding approved sources to Group Policy. To get the same values with PowerShell the example blow would perform the same behavior.
+
+
+```PowerShell
+Get-WinGetSource | convertto-json -compress
+```
+
+---
+
+### Client command `winget search`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Find-WinGetPackage
+```
+
+**Parameters**
+
+-PackageIdentifier
+
+-Name
+
+-Moniker
+
+-Tag
+
+-Command
+
+-Source
+
+-Count
+
+-Exact
+
+---
+
+### Client command `winget list`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Get-WinGetPackage
+```
+**Parameters**
+
+-PackageIdentifier
+
+-Name
+
+-Moniker
+
+-Source
+
+-Tag
+
+-Command
+
+-Count
+
+-Exact
+
+---
+
+### Client command `winget upgrade`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Update-WinGetPackage
+```
+**Parameters**
+
+-Manifest  
+
+-PackageIdentifier
+
+-Name
+
+-Moniker
+
+-Version
+
+-Source
+
+-Exact
+
+-Interactive
+
+-Silent
+
+-Log
+
+-Override
+
+-Location
+
+-All
+
+---
+
+### Client command `winget uninstall`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Uninstall-WinGetPackage
+```
+
+**Parameters**
+
+-Manifest  
+
+-PackageIdentifier
+
+-Name
+
+-Moniker
+
+-Version
+
+-Source
+
+-Exact
+
+-Interactive
+
+-Silent
+
+-Log
+
+**Example**
+
+>Note: The example below could be executed to remove all packages with "xbox" in their name, or in other metadata fields provided in the manifest (assumes there is a match with the installed package in a configured source).
+
+```PowerShell
+Get-WingetPackage xbox | UninstallWingetPackage
+```
+
+---
+
+### Client command `winget hash`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Get-WinGetHash
+```
+
+**Parameters**
+
+-Installer
+
+-Signature
+
+---
+
+### Client command `winget validate`
+
+**PowerShell Cmdlet**
+
+Test-WinGetManifest
+
+> Note: we are considering building an "interactive" validation that could test installing the package in the Windows Sandbox for example. That could be another cmdlet or a switch like "-validate".
+
+---
+
+### Client command `winget settings`
+
+**PowerShell Cmdlets**
+
+```PowerShell
+Get-WinGetSetting
+```
+>Note: This will not open the settings file. It will just get the object.
+
+**Parameters**
+
+-Path
+>Note: This would provide the path to the settings.json file.
+
+```PowerShell
+Set-WinGetSetting
+```
+
+> Note: This would apply the settings provided by the object built or modified by the user.
+
+**Parameters**
+
+-File
+
+---
+
+### Client command `winget features`
+
+**PowerShell Cmdlets**
+
+```PowerShell
+Get-WinGetFeature
+```
+
+```PowerShell
+Enable-WingetFeature
+```
+
+```PowerShell
+Disable-WingetFeature
+```
+
+---
+
+### Client command `winget export`
+
+**PowerShell Cmdlet**
+
+```PowerShell
+Export-WinGetPackageSet 
+```
+**Parameters**
+
+-Output
```

Comment:
`-Path` and `-LiteralPath` might be better parameter names for `-Output`. See [Export-Csv](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/export-csv?view=powershell-7.1)

Reply:
@megamorf what is the difference between the two?

Reply:
Per convention, `-Path` accepts wildcards and globs. LiteralPath does not. There's probably official docs for this as it's very standard

Reply:
Correct. The official documentation says the following:
> Unlike **Path**, the value of the **LiteralPath** parameter is used exactly as it is typed. No characters are interpreted as wildcards. If the path includes escape characters, use single quotation marks. Single quotation marks tell PowerShell not to interpret any characters as escape sequences.

Why is that important? Let's consider that you have a folder with square brackets or an asterisk in its name. `-Path` will treat that as an expression rather than a true representantion of the path. Users have learned to use `-LiteralPath` when they literally want PowerShell to treat the user specified path as it is.

Reply:
OK, so rather than "-Output" I believe what I'm reading is we should support both "-Path" and "-LiteralPath". Is it safe to assume PowerShell handles the wildcards when "-Path" is used?

Reply:
So far as I can recall, -Path takes a wildcarded string and the cmdlet does the variable expansion.